Text/Distinct/Thesaurus.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) Microsoft. All rights reserved.  
+// Licensed under the MIT License. See LICENSE file in the project root for full license information.  
+
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace AzureCognitiveSearch.PowerSkills.Text.Distinct
+{
+    public class Thesaurus
+    {
+        public Thesaurus(string executingDirectoryPath)
+            : this(JsonConvert.DeserializeObject<IEnumerable<IEnumerable<string>>>(
+                File.ReadAllText($"{executingDirectoryPath}\\thesaurus.json")))
+        { }
+
+        public Thesaurus(IEnumerable<IEnumerable<string>> dataset)
+        {
+            Synonyms = new Dictionary<string, string>(StringComparer.InvariantCultureIgnoreCase);
+            foreach (IEnumerable<string> lemma in dataset)
+            {
+                string canonicalForm = lemma.First();
+                foreach (string form in lemma)
+                {
+                    Synonyms.Add(Normalize(form), canonicalForm);
+                }
+            }
+        }
+
+        public Dictionary<string, string> Synonyms
+        {
+            get; private set;
```

Comment:
>private set; [](start = 17, length = 12)

nit: doesn't look like you need a private set - its only assigned in constructor